Client/Frontend/Browser/SearchEngines/DefaultSearchEngineProvider.swift

```diff
@@ -0,0 +1,101 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import UIKit
+import Shared
+
+protocol SearchEngineProvider {
+    func getOrderedEngines(customEngines: [OpenSearchEngine],
+                           orderedEngineNames: [String]?,
+                           completion: @escaping ([OpenSearchEngine]) -> Void)
+    func getUnorderedBundledEnginesFor(locale: Locale,
+                                       possibleLanguageIdentifier: [String],
+                                       completion: @escaping ([OpenSearchEngine]) -> Void)
+}
+
+class DefaultSearchEngineProvider: SearchEngineProvider {
+    private let orderedEngineNames = "search.orderedEngineNames"
+
+    func getOrderedEngines(customEngines: [OpenSearchEngine],
```

Comment:
This is only the refactor right?

Reply:
Yup, some functionality moved out of SearchEngines and into here for testability.